src/appengine/handlers/cron/group_leader.py

```diff
@@ -62,14 +66,35 @@ def _key_func(testcase):
 
     items = sorted(items, reverse=True, key=lambda t: t.timestamp)
 
-    for item in items:
+    asan_index = None
+    security_index = None
+    i386_indexes = []
+    for idx, item in enumerate(items):
       item.is_leader = False
-
-    leader_index = find_index(items, is_reproducible_and_has_issue)
+      if not security_index and item.security_flag:
+        security_index = idx
+      try:
+        testcase = data_handler.get_testcase_by_id(item.id)
```

Comment:
Can we avoid the `get_testcase_by_id` here? That may slow things down. If needed, we can add the needed attribute (`job_type`) to `FORWARDED_ATTRIBUTES` in `grouper` instead to avoid additional queries.

Reply:
yes that makes things easier!

Reply:
will pick it up tmrw.

Reply:
Done.